### DISCUSSION TITLE
Disable the `no-negated-condition` rule

### DISCUSSION THREAD

**Discussion by leroykorterink** [mediamonks/eslint-config#223] [General]
There is a proposal to disable the `no-negated-condition` rule.

> `no-negated-condition`: this rule is one of the top contenders for the most annoying rule. Sometimes it's easier to read a negated condition than the positive one.
> \- @edgardz

**Comment by leroykorterink** [comment]
Negated conditions are more difficult to understand. Code can be made more readable by inverting the condition instead. Disallowing negated conditions will also prevent unnecessary logical operations.

Incorrect code:

```ts
if (!a) {
  c();
} else {
  b();
}

if (a !== b) {
  c();
  } else {
  b();
}

!a ? c : b
``` 

Correct code:

```ts
if (a) {
  b();
} else {
  c();
}

if (a === b) {
  b();
  } else {
  c();
}

a ? b : c
``` 

Please see [the documentation](https://eslint.org/docs/latest/rules/no-negated-condition) of the rule.

**Reply by ThaNarie** [reply]
The examples in the docs don't give a good example of common use cases, where your variable and function names might already contain positive or negative names (`isEnabled` vs `isDisabled`).

I also suspect that the main issue with this rule, is what the "main task" is, that you want in the `if`, and what the "other less important thing" is, that you might want in the `else`.

So I'm very curious to see some code examples.
